### PR TITLE
Add haskell-lexeme.el

### PR DESCRIPTION
**Pull request by gracjan**
Implement Haskell tokenizer, with all bells a whistles, Unicode support, all GHC special cases and stuff.

Related to #724.

### CODE REVIEW

**Comment by gracjan** [comment]
@kuribas, @geraldus: can you have a look if you see something obvious missing here?

File: haskell-lexeme.el [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      'conid)
+     ((or (equal char ?_)
+          (member category '(Ll Lo)))
+      'varsym)
+     ((and (>= char ?0) (<= char 9))
+      'number)
+     ((member char '(?\] ?\[ ?\( ?\) ?\{ ?\} ?\` ?\, ?\;))
+      'special))))
+
+(defun haskell-utils-match-string-equal (string &optional index)
```
Comment (geraldus) [inline haskell-lexeme.el]: would `haskell-utils.el` be the better place for this function?

**Comment by geraldus** [comment]
@gracjan I keep an eye on every pull request and issue, and this is no exception.  This looks good to me, but my level of Haskell experience does not allow me to give any valuable comments on this (besides small notice I've already put).  Overall, I think this is very important for present and future development.  Good stuff.

**Comment by gracjan** [comment]
Thanks for the feedback.

Is there any other test that you can think of and should be implemented?

**Pull request merged**